viewer/packages/360-images/src/entity/Image360RevisionEntity.ts

```diff
@@ -63,47 +61,51 @@ export class Image360RevisionEntity implements Image360Revision {
-   * and one promise for when both downloads are completed. If the low resolution images are completed first the full resolution
-   * download and texture loading will continue in the background, and applyFullResolution can be used to apply full resolution textures
-   * to the image360VisualzationBox at a desired time.
+   * This will start the download of both low and full resolution textures and return one promise for when each texture set is ready.
+   * If the low resolution images are completed first the full resolution download and texture loading will continue in the background,
+   * and applyFullResolutionTextures can be called to apply full resolution textures to the image360VisualzationBox at a desired time.
    *
-   * @returns firstCompleted A promise for when the first set of textures has been loaded.
-   * @returns fullResolutionCompleted A promise for when full resolution images are done loading.
+   * @returns lowResolutionCompleted A promise for when the low resolution images are done loading.
+   * @returns fullResolutionCompleted A promise for when the full resolution images are done loading.
    */
   public loadTextures(abortSignal?: AbortSignal): {
-    firstCompleted: Promise<void>;
+    lowResolutionCompleted: Promise<void>;
     fullResolutionCompleted: Promise<void>;
   } {
-    const lowResolutionFaces = this._imageProvider
-      .getLowResolution360ImageFiles(this._image360Descriptor.faceDescriptors, abortSignal)
-      .then(async faces => {
-        return { textures: this._image360VisualzationBox.loadFaceTextures(faces), isLowResolution: true };
-      });
-
-    const fullResolutionFaces = this._imageProvider
-      .get360ImageFiles(this._image360Descriptor.faceDescriptors, abortSignal)
-      .then(async faces => {
-        return { textures: this._image360VisualzationBox.loadFaceTextures(faces), isLowResolution: false };
-      });
+    const lowResolutionCompleted = this.loadPreviewTextures(abortSignal);
+    const fullResolutionCompleted = this.loadFullTextures(abortSignal);
+
+    this._onFullResolutionCompleted = fullResolutionCompleted;
 
     this.loadAnnotations();
 
-    const firstCompleted = Promise.any([lowResolutionFaces, fullResolutionFaces]).then(
-      async ({ textures, isLowResolution }) => {
-        this._textures = await textures;
+    return { lowResolutionCompleted, fullResolutionCompleted };
+  }
 
-        if (isLowResolution) {
-          this._getFullResolutionTextures = fullResolutionFaces;
-        }
-      }
+  private async loadPreviewTextures(abortSignal?: AbortSignal): Promise<void> {
+    const previewImageFiles = await this._imageProvider.getLowResolution360ImageFiles(
+      this._image360Descriptor.faceDescriptors,
+      abortSignal
     );
+    const textures = await this._image360VisualzationBox.loadFaceTextures(previewImageFiles);
+    if (this._textures.length === 6) {
```

Comment:
This `if` statement should be a little more clear, probably renaming `textures` can help.